src/modules/git_status.rs

```diff
@@ -18,48 +20,23 @@ use super::{Context, Module};
 ///   - `»` — A renamed file has been added to the staging area
 ///   - `✘` — A file's deletion has been added to the staging area
 pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
-    // This is the order that the sections will appear in
-    const GIT_STATUS_CONFLICTED: &str = "=";
-    const GIT_STATUS_AHEAD: &str = "⇡";
-    const GIT_STATUS_BEHIND: &str = "⇣";
-    const GIT_STATUS_DIVERGED: &str = "⇕";
-    const GIT_STATUS_UNTRACKED: &str = "?";
-    const GIT_STATUS_STASHED: &str = "$";
-    const GIT_STATUS_MODIFIED: &str = "!";
-    const GIT_STATUS_ADDED: &str = "+";
-    const GIT_STATUS_RENAMED: &str = "»";
-    const GIT_STATUS_DELETED: &str = "✘";
-    const PREFIX: &str = "[";
-    const SUFFIX: &str = "] ";
-
     let repo = context.get_repo().ok()?;
     let branch_name = repo.branch.as_ref()?;
     let repo_root = repo.root.as_ref()?;
     let repository = Repository::open(repo_root).ok()?;
 
     let mut module = context.new_module("git_status");
-    let show_sync_count = module.config_value_bool("show_sync_count").unwrap_or(false);
-    let module_style = module
-        .config_value_style("style")
-        .unwrap_or_else(|| Color::Red.bold());
-    let start_symbol = module
-        .config_value_str("prefix")
-        .unwrap_or(PREFIX)
-        .to_owned();
-    let end_symbol = module
-        .config_value_str("suffix")
-        .unwrap_or(SUFFIX)
-        .to_owned();
+    let config: GitStatusConfig = GitStatusConfig::try_load(module.config);
 
     module
         .get_prefix()
-        .set_value(start_symbol)
-        .set_style(module_style);
+        .set_value(config.prefix)
+        .set_style(config.style);
     module
         .get_suffix()
-        .set_value(end_symbol)
-        .set_style(module_style);
-    module.set_style(module_style);
+        .set_value(config.suffix)
+        .set_style(config.style);
+    module.set_style(config.style);
```

Comment:
`module.set_style` twice?

Reply:
The first style is for the suffix, and the second is for the module, I think.

Reply:
Yeah